spec/optional_query_cache_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'action_controller/metal/exceptions'
+require 'active_record'
+require 'rack'
+
+require 'spec_helper'
+
+require 'optional_query_cache'
+
+RSpec.describe OptionalQueryCache::QueryCache do
+  let(:app) { ->(env) { [200, env, 'app'] } }
+
+  let :subject do
+    described_class.new(app)
+  end
+
+  let :env do
+    double.tap do |env|
+      allow(env).to receive(:[]).with('REQUEST_PATH').and_return('')
+    end
+  end
+
+  it 'permits query caching by default' do
+    expect(ActiveRecord::Base).to receive(:connection).and_return(
```

Comment:
two expectations in the same example is usually a bad sign - is this really an expectation, or a stub?